### ISSUE TITLE
Access errors for a specific field?

### ISSUE THREAD

**Issue by Skyliegirl33** [jprochazk/garde#46]
Noticed this crate just recently. I've been using `validator` but ran into some limitations.
Say you have a nested structure like so:
```rust
#[derive(Debug, Validate, Serialize, Deserialize, Clone)]
pub struct UserIdentifier {
  #[validate(range(max = 4, message = "Identifier is too long"))]
  pub id: usize
}

#[derive(Debug, Validate, Serialize, Deserialize, Clone)]
pub struct UserRole {
  #[validate(length(min = 4, message = "Role name too short"))]
  pub name: String,
  #[validate]
  pub identifiers: Vec<UserIdentifier>
}
```
It would be neat if there's any possibility to access the errors of a specific field like: `errors.identifiers[0].id` or `errors.name`, which would return an array of issues for that field, sort of like Zod's [format()](https://zod.dev/ERROR_HANDLING?id=formatting-errors).

However, I'm not sure if this is possible to do in Rust. It would be cool to have easier access to the individual field errors, though.

**Comment by jprochazk** [comment]
The error type is already a defined as recursive enum, so you should be able to match on it to some extent https://github.com/jprochazk/garde/blob/719c7f8bf8dfa6fbd11c9c14eeb793ce1b448e3d/garde/src/error.rs#L44-L61

**Comment by Skyliegirl33** [comment]
That's true, however I was looking for something a little more ergonomic to use.

**Comment by jprochazk** [comment]
I added a `select` macro which should address this
```rust
use garde::Validate;

#[derive(Validate)]
pub struct UserIdentifier {
    #[garde(range(max = 4))]
    pub id: usize,
}

#[derive(Validate)]
pub struct UserRole {
    #[garde(ascii, byte_length(min = 10))]
    pub name: String,
    #[garde(dive)]
    pub identifiers: Vec<UserIdentifier>,
}

let v = UserRole {
    name: "😂".into(),
    identifiers: vec![UserIdentifier { id: 10 }, UserIdentifier { id: 20 }],
};

let report = v.validate(&()).unwrap_err();
assert_eq!(
    garde::select!(report, identifiers[0])
        .map(|e| e.to_string())
        .collect::<Vec<String>>(),
    ["greater than 4"]
);
assert_eq!(
    garde::select!(report, name)
        .map(|e| e.to_string())
        .collect::<Vec<String>>(),
    ["not ascii", "byte length is lower than 10"]
);
```

Will be released as part of `v0.15`.
